release/release.go

```diff
@@ -35,47 +37,33 @@ type OS struct {
 	VersionID string `json:"version-id,omitempty"`
 }
 
+var (
+	versionSignaturePath = "/proc/version_signature"
+	apparmorSysPath      = "/sys/kernel/security/apparmor/"
+)
+
 // ForceDevMode returns true if the distribution doesn't implement required
 // security features for confinement and devmode is forced.
-func (os *OS) ForceDevMode() bool {
-	switch os.ID {
-	case "neon":
-		return false
-	case "ubuntu":
-		return false
-	case "ubuntu-core":
-		return false
-	case "elementary":
-		switch os.VersionID {
-		case "0.4":
-			return false
-		}
-	case "linuxmint":
-		// NOTE: mint uses "LinuxMint" (mixed capitalization) but this is
-		// normalized by readOSRelease.
-		switch os.VersionID {
-		case "18.1":
-			// Linux Mint 18.1 aka "serena" should use apparmor confinement
-			// given that it shares packages with Ubuntu 16.04.
-			return false
-		}
-	case "galliumos":
-
-		switch os.VersionID {
-		case "1.0":
-			// 1.0 is based on 15.04
-			return true
+func (o *OS) ForceDevMode() bool {
+	// Check if kernel signature contains "Ubuntu", currently only
+	// the Ubuntu kernels have all the required apparmor patches
+	// (but those are getting upstreamed so at some point we need
+	// to make this check smater)
+	versionSig, err := ioutil.ReadFile(versionSignaturePath)
+	if err != nil {
+		return true
```

Comment:
Maybe add a comment that explains this file is optional and we don't care abut  the error. You may also check if `...insert go code that checks if error is ENOENT` but not sure if relevant.